include/onnxruntime/core/session/onnxruntime_cxx_api.h

```diff
@@ -1051,6 +1054,54 @@ struct ArenaCfg : Base<OrtArenaCfg> {
 // Custom OPs (only needed to implement custom OPs)
 //
 
+/// <summary>
+/// This struct wraps a raw pointer const OrtNodeArg* that is obtained from
+/// a KernelInfo object. Use it to access the name or type info of an input
+/// or output.
+/// </summary>
+struct NodeArg {
+  explicit NodeArg(const OrtNodeArg* node_arg);
+
+  // Returns a pair consisting of a Ort::AllocatedStringPtr and the string's length.
+  // Ex: auto name = node_arg.GetName(allocator);
+  // name.first.get(); // Get char*
+  // name.second; // Get length
+  std::pair<AllocatedStringPtr, size_t> GetName(OrtAllocator* allocator) const;
```

Comment:
>GetName(OrtAllocator* allocato [](http://example.com/codeflow?start=40&length=30)

Allocating names using ORT allocator is a strange practice. Certainly should not come up to CPP level. Make a copy of it to std::string and deallocate within the CPP wrapper so user does not have these headaches. Need to discuss.

Reply:
I see. I've saw that previous APIs handled returning names in various ways, and I wasn't quite sure which was the current accepted practice.

One method (A) requires the user to first query for the name's size (by passing nullptr), then the user allocates memory, and finally calls again to have the API fill in the name. The other method (B) just passes in an allocator, which the user must use to free the data.

I can change this to use method A.